support/util/networking.go

```diff
@@ -121,3 +130,56 @@ func AllowedCIDRBlocks(hcp *hyperv1.HostedControlPlane) []hyperv1.CIDRBlock {
 	}
 	return nil
 }
+
+func SetAdvertiseAddresses(hcp *hyperv1.HostedControlPlane, defaults *DefaultAdvIps) []string {
+	addresses := make([]string, 0)
+
+	// User defined AdvertiseAddress/AdvertiseAddresses
+	if hcpAdvAddresses := AdvertiseAddresses(hcp); hcpAdvAddresses != nil {
```

Comment:
Why is this if needed? why don't we just `addresses = AdvertiseAddresses(hcp)`?
then only if addresses is still empty you run AdvertiseAddress(hcp), then only if adresses stil empty you return default.

Reply:
Makes sense. I've modified the function in that way and also I've added some UnitTests over it because I think they was necessary.